### ISSUE TITLE
Makefile.pl

### ISSUE THREAD

**Issue by xFFFFF** [taskula/binance-perl-api#1]
Could You prepare makefile.pl or some installation instructions?

**Comment by ibrierley** [comment]
Whilst waiting for instructions, if you are struggling to get it going...try extracting the zip to a folder, and then create a file like below and see if it runs......note, it's assuming there is a lib folder alongside it, which has the Binance folder inside that. Naturally you can move it where you want, but make sure the top line references any extra lib files.

You don't need a key to test the bit below.

#!/usr/bin/perl -I lib

use strict;
use warnings;

use Binance::API;
use Data::Dumper;

my $api = Binance::API->new(
    apiKey    => 'my_api_key',
    secretKey => 'my_secret_key',
);


my $ticker = $api->ticker( symbol => 'ETHBTC' );

warn Dumper $ticker;

**Comment by taskula** [comment]
Thanks for the interest. I've had this project on hold for a couple of months now. I'll provide installation instructions.

As you may have noticed, not all Binance API actions are implemented. Feel free to provide extensions to our current implementation.

**Comment by xFFFFF** [comment]
@taskula are You sure that signature for "query" is generating correctly? I tried add allOrders query support in API.pm but had error from binance:
```
[Binance::API::Request::_exec] New request: https://api.binance.com/api/v3/allOrders?symbol=TRXBTC&timestamp=1526670495159 at /home/f/perl5/lib/perl5/Binance/API/Request.pm line 75.
[Binance::API::Request::_exec] Bad Request at /home/f/perl5/lib/perl5/Binance/API/Request.pm line 93.
SUB: Binance::API::Request::_execSUB: Binance::API::Request::_exec{"code":-1102,"msg":"Mandatory parameter 'signature' was not sent, was empty/null, or malformed."}
-------------
Use of uninitialized value $message in concatenation (.) or string at /home/f/perl5/lib/perl5/Binance/API/Logger.pm line 54.
[(eval)]  at binanceapi.pl line 24.
SUB: (eval)
```
I added to API.pm these lines:
```
sub allorders {
    my ($self, %params) = @_;
    unless ($params{'symbol'}) {
        $self->log->error('Parameter "symbol" required');
        Binance::Exception::Parameter::Required->throw(
            error => 'Parameter "symbol" required',
            parameters => ['symbol']
        );
    }
    my $query = {
        symbol    => $params{'symbol'},
        orderId => $params{'orderId'},
        limit => $params{'limit'},
    };
    return $self->ua->get('/api/v3/allOrders', { signed => 1, query => $query } );
}
```
I would love to help with the development of this project, but this problem limits me. Would you be so kind and directed me what should I do?

For GET server dont allow the use requestbody.

Thanks for great work! When I analyze the code (and I'm still learning Perl) I come to the conclusion that if I had to write it myself it would take forever.

**Comment by taskula** [comment]
There seems to be a bug. Thanks for noticing, I'll fix it!

**Comment by taskula** [comment]
@xFFFFF 

As of now I found at least two issues (fixed in 98e3a68401dfbf09388729cf4918fc3d0ec1e277):
1. I was using SHA512 instead of SHA256
2. Signature was not included in the GET request with only query parameters

I now get another error:
`{"code":-1022,"msg":"Signature for this request is not valid."}`

If you want to figure out what's wrong with the signature calculation, feel free :)

I'm sure we can work this out together! See issue #3
